receipt_upload/receipt_upload/merchant_resolution/embedding_processor.py

```
@@ -183,26 +282,108 @@ def process_embeddings(
+                # Wait for both to complete
+                for future in as_completed([merchant_future, validation_future]):
+                    try:
+                        future.result()  # Raise any exceptions
+                    except Exception as e:
+                        _log(f"WARNING: Parallel task failed: {e}")
+                        logger.exception("Parallel task failed")
+
+                # Get results (may raise if failed)
+                try:
+                    merchant_result = merchant_future.result()
+                except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)